packages/amplify-cli/src/commands/upgrade.ts

```diff
@@ -0,0 +1,65 @@
+import { $TSContext, isPackaged, pathManager } from 'amplify-cli-core';
+import execa from 'execa';
+import fetch from 'node-fetch';
+import { gt } from 'semver';
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import ora from 'ora';
+import { oldVersionPath } from '../utils/win-constants';
+import chalk from 'chalk';
+
+const binUrl = (version: string, platform: 'macos' | 'win.exe' | 'linux') =>
+  `https://github.com/aws-amplify/amplify-cli/releases/download/v${version}/amplify-pkg-${platform}`;
+const latestVersionUrl = 'https://api.github.com/repos/aws-amplify/amplify-cli/releases/latest';
+
+export const run = async (context: $TSContext) => {
+  if (!isPackaged) {
+    context.print.warning(
+      `"upgrade" is not supported in this installation of Amplify.\nUse ${chalk.blueBright('npm i -g @aws-amplify/cli')}`,
+    );
+    return;
+  }
+  const { version: thisVersion } = require('../../package.json');
+  if (typeof thisVersion !== 'string') {
+    throw new Error('Cannot determine current CLI version. Try uninstalling and reinstalling the CLI.');
+  }
+  const latestVersion = await getLatestVersion();
+  if (gt(latestVersion, thisVersion)) {
+    await upgradeCli(latestVersion);
+    context.print.success(`Successfully upgraded to Amplify CLI version ${latestVersion}!`);
+  } else {
+    context.print.info('This is the latest Amplify CLI version.');
+  }
+};
+
+const upgradeCli = async (version: string) => {
+  const isWin = process.platform.startsWith('win');
+  const binPath = path.join(pathManager.getHomeDotAmplifyDirPath(), 'bin', isWin ? 'amplify.exe' : 'amplify');
+  const platformSuffix = isWin ? 'win.exe' : process.platform === 'darwin' ? 'macos' : 'linux';
+  const url = binUrl(version, platformSuffix);
+  const spinner = ora();
+
+  if (isWin) {
+    await fs.move(binPath, oldVersionPath);
```

Comment:
Is there a flow in which we need to keep the old ver on win? It looks like we delete it every time the CLI is executed.

Reply:
No we don't need to keep the old version